applications/StructuralMechanicsApplication/custom_conditions/displacement_shifted_boundary_condition.cpp

```diff
@@ -0,0 +1,540 @@
+    const SizeType n_dim = rCurrentProcessInfo[DOMAIN_SIZE];
+    const SizeType local_size = n_dim * n_nodes;
+    if (rConditionalDofList.size() != local_size){
+        rConditionalDofList.resize(local_size);
+    }
+
+    // Fill the DOFs vector from the condition nodes
+    if (n_dim == 2) {
+        for (std::size_t i = 0; i < n_nodes; ++i) {
+            rConditionalDofList[i*n_dim] = r_geometry[i].pGetDof(DISPLACEMENT_X);
+            rConditionalDofList[i*n_dim + 1] = r_geometry[i].pGetDof(DISPLACEMENT_Y);
+        }
+    } else {
+        for (std::size_t i = 0; i < n_nodes; ++i) {
+            rConditionalDofList[i*n_dim] = r_geometry[i].pGetDof(DISPLACEMENT_X);
+            rConditionalDofList[i*n_dim + 1] = r_geometry[i].pGetDof(DISPLACEMENT_Y);
+            rConditionalDofList[i*n_dim + 2] = r_geometry[i].pGetDof(DISPLACEMENT_Z);
+        }
+    }
+
+        KRATOS_CATCH("")
+    }
+
+void DisplacementShiftedBoundaryCondition::Initialize(const ProcessInfo &rCurrentProcessInfo)
+{
+    // Get the parent constitutive law from the properties and clone them to be current properties
+    // Note that we are assuming that the parent properties were assigned as properties when creating the condition
+    const auto& r_prop = GetProperties();
+    const auto p_cons_law = r_prop[CONSTITUTIVE_LAW];
+    const auto& r_N = this->GetValue(SHAPE_FUNCTIONS_VECTOR);
+    if (p_cons_law != nullptr) {
+        this->SetValue(CONSTITUTIVE_LAW, p_cons_law->Clone());
+        this->GetValue(CONSTITUTIVE_LAW)->InitializeMaterial(r_prop, GetGeometry(), r_N);
+    } else {
+        KRATOS_ERROR << "A constitutive law needs to be specified in condition " << this->Id() << "." << std::endl;
+    }
+}
+
+void DisplacementShiftedBoundaryCondition::CalculateLocalSystem(
+    MatrixType& rLeftHandSideMatrix,
+    VectorType& rRightHandSideVector,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    // Get problem dimensions
+    const SizeType n_dim = rCurrentProcessInfo[DOMAIN_SIZE];
+
+    // Check (and resize) LHS and RHS matrix
+    const auto &r_geometry = this->GetGeometry();
+    const std::size_t n_nodes = r_geometry.PointsNumber();
+    const std::size_t local_size = n_nodes * n_dim;
+    if (rRightHandSideVector.size() != local_size) {
+        rRightHandSideVector.resize(local_size, false);
+    }
+    if (rLeftHandSideMatrix.size1() != local_size || rLeftHandSideMatrix.size2() != local_size) {
+        rLeftHandSideMatrix.resize(local_size, local_size, false);
+    }
+
+    // Set LHS and RHS to zero
+    noalias(rRightHandSideVector) = ZeroVector(local_size);
+    noalias(rLeftHandSideMatrix) = ZeroMatrix(local_size, local_size);
+
+    // Get meshless geometry data
+    const double w = GetValue(INTEGRATION_WEIGHT);
+    const auto& r_N = GetValue(SHAPE_FUNCTIONS_VECTOR);
+    const auto& r_DN_DX = GetValue(SHAPE_FUNCTIONS_GRADIENT_MATRIX);
+    array_1d<double,3> normal = GetValue(NORMAL);
```

Comment:
I would add a check of the norm being different of 0

Reply:
Fair enough. But rather than putting it in here (it will be evaluated at each iteration), I'd put it in the `ShiftedBoundaryMeshlessInterfaceUtility` that calculates the normal values, so we only check it once.